app/resource.py

```diff
@@ -0,0 +1,357 @@
+    def get(self, id):
+        """
+        This endpoint returns list of bucketlist items based on access token and bucketlist id.
+        Method: GET
+        Parameters:
+            limit (optional)    default=25
+            page  (optional)    default=1
+        Header:   
+            AccessToken  (required)
+
+        Response: JSON
+        """
+
+        user_id = get_user_id_from_token(token)
+        bucketlist = BucketListModel.query.filter_by(id=id, created_by=int(user_id)).first()
+        if not bucketlist:
+            abort(403, message="Unauthorized access")
+
+        limit = int(request.args.get('limit',25))
+        page = int(request.args.get('page',1))
+        q = request.args.get('q','')
+        offset = (page * limit) - limit
+        user_id = get_user_id_from_token(token)
+        next_page = page + 1
+        prev_page = page - 1 if page > 1 else 1
+        all_items = BucketListItemModel.query.filter(
+            BucketListItemModel.task.like("%" + q + "%"), 
+            BucketListItemModel.bucketlist_id.is_(bucketlist.id)
+            ).all()
+        items = all_items[offset:(limit + offset)]
+        result = {'data':[item.get() for item in items],'page':{}}
+        if (limit + offset) < len(all_items):
+            result['page']['next'] = "/api/v1.0/bucketlists/" + str(id) + "?limit=" + str(limit) + "&page=" + str(next_page)
+        
+        if offset:
+            result['page']['prev'] = "/api/v1.0/bucketlists" + str(id) + "?limit=" + str(limit) + "&page=" + str(prev_page)
+
+        return result
+
+    @auth.login_required
+    def post(self, id):
+        """
+        This endpoint returns created bucketlist item details.
+        Method: POST
+        Parameters:
+            task (required) 
+        Header:   
+            AccessToken  (required)
+
+        Response: JSON
+        """
+
+        args = validate_args({'task':True})
+        user_id = get_user_id_from_token(token)
+        bucketlist = BucketListModel.query.filter_by(id=id, created_by=int(user_id)).first()
+        if not bucketlist:
+            abort(403, message="Unauthorized access")
+        
+        item = BucketListItemModel(args['task'], bucketlist)    
+        if not save(item):
+            abort(409, message="Item already exists")
+
+        return {'data': item.get()}, 201
+
+
+
+class Login(Resource):
+    def post(self):
+        """
+        This will authenticate user and provide token used to access other resources
+
+        Method: POST
+        Parameters: 
+            username    (required)
+            password    (required)
+        Response: JSON
+        """
+        
+        args = validate_args({'username':True, 'password':True})
+        user = User.query.filter_by(username=args['username'],password=md5(args['password'])).first()
+
+        if not user:
+            abort(403, message='Invalid user credentials')
+
+        return {'token': user.generate_token(),'data': user.get()}, 200
```

Comment:
**Code Quality: Nitpick**

There should be no wrapper for your returned data. For this all that is required is the `token` data.